AEA Web.js

```diff
@@ -36,99 +36,168 @@
+	var found = false;
+	var rows = ZU.xpath(doc, '//article//a[contains(@href, "/articles?id=")]|//li[@class="article"]//a[contains(@href, "/articles?id=")]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
 }
 
 function doWeb(doc, url) {
-	var arts = new Array();
 	if (detectWeb(doc, url) == "multiple") {
-		var items = new Object();
-		var title;
-
-		var titles = doc.evaluate('//a[contains(@href, "articles.php?doi") and @style="font-weight:bold;"]', doc, null, XPathResult.ANY_TYPE, null);
-		while (title = titles.iterateNext()) {
-			items[title.href] = title.textContent;
-		}
-
-		Zotero.selectItems(items, function(items) {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
 			if (!items) {
 				return true;
 			}
+			var articles = new Array();
 			for (var i in items) {
-				arts.push(i);
+				articles.push(i);
 			}
-			Zotero.Utilities.processDocuments(arts, scrape);
+			ZU.processDocuments(articles, scrape);
 		});
 	} else {
 		scrape(doc, url);
 	}
 }
 
 function scrape(doc, url) {
-		var risURL = url.replace(/articles\.php\?/, "content/articles/include/file_export.php?") + "&format=ris";
-		Z.debug(risURL)
-		var abstract = ZU.xpathText(doc, '//div[@class="sub_head_dialog" and contains(text(), "Abstract")]/following-sibling::div[1]');
-		//Z.debug(abstract)
-		ZU.HTTP.doGet(risURL, function(text) {
-			//Z.debug(text)
-			text = text.trim()
-			var translator = Zotero.loadTranslator("import");
-			translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
-			translator.setString(text);
-			translator.setHandler("itemDone", function(obj, item) {
-				item.abstractNote = abstract;
-				var pdfURL = "https://www.aeaweb.org/atypon.php?return_to=/doi/pdfplus/" + item.DOI;
-				//Z.debug(pdfURL)
-				//Remove period at end of title
-				item.attachments.push({
-					url: pdfURL,
-					title: "AEAweb Full Text PDF",
-					mimeType: "application/pdf"
-				});
-				item.complete();
-			});
-			translator.translate();
+	var translator = Zotero.loadTranslator('web');
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');//Embedded Metadata
+	translator.setDocument(doc);
+	translator.setHandler("itemDone", function(obj, item) {
+		//Correct pages format, e.g. 1467-96 or 625-63
+		var m = item.pages.match(/^(\d+)(\d\d)[\--](\d\d)$/);
+		if (m.length > 0) {
+			item.pages = m[1]+m[2]+"-"+m[1]+m[3];
+		}
+		
+		//The abstract is contained in the section-node of class abstract,
+		//but this node consists of an (empty) text node, a h2 node
+		//and another text node with the actual abstract.
+		var abstract = ZU.xpathText(doc, '//section[contains(@class,"abstract")]/text()[last()]');
+		item.abstractNote = abstract;
+		
+		var pdfURL = "https://www.aeaweb.org/atypon.php?doi=" + item.DOI;
```

Comment:
from the tests it looks like you don't need that since EM already attacheds a PDF? Or does that one not work? In that case, overwrite it instead of pushing.